### DISCUSSION TITLE
Support for Grouped (?) Module Structure

### DISCUSSION THREAD

**Discussion by m-kas** [nWidart/laravel-modules#1986] [General]
Hi,

I'm trying to organize my modules using a nested folder structure like this:

```
Modules/
├── Core/
│   ├── Users
│   ├── Calculations
└── Feature/
    ├── ModuleFeatureOne
    ├── AnotherFeature
```

I tried different approaches like:
`php artisan module:make Users Core/Users`
and also modified `config/modules.php` in different ways but I was not able to achieve my intended goal.

Is it possible to structure modules in nested directories like Core and Feature using your package? If so, could you provide guidance on the best way to configure this?

**Comment by dcblogdev** [comment]
sorry its not supported. What I've done in the past is add additional scan directories into the modules.php config file:

```php
'scan' => [
        'enabled' => true,
        'paths'   => [
            base_path('ModulesCms'),
        ],
    ],
```

Then moved modules into this directory you will have to manually change the namespaces plus any commands to generate files would expect the default location so may run intro trouble there.

I've only done this way once. there's been a few attempts to add support for nested modules but currently its not supported.

**Comment by m-kas** [comment]
Thanks for your answer. Than I will simply go with prefix in the module names to separate my modules, like `Core…` and `Feature…`, because I really need this division.
